doc/source/reference/c-api/iterator.rst

```diff
@@ -203,6 +203,69 @@ is used to control the memory layout of the allocated result, typically
     }
 
 
+Multi Index Tracking Example
+----------------------------
+
+This example shows you how to work with the :c:data:`NPY_ITER_MULTI_INDEX` flag. For simplicity, we assume the argument is a two-dimensional array.
+
+.. code-block:: c
+
+   void PrintMultiIndex(PyArrayObject *arr) {
+     NpyIter *iter;
+     NpyIter_IterNextFunc *iternext;
+     char **dataptr;
+
+     iter = NpyIter_New(
+         arr, NPY_ITER_READONLY | NPY_ITER_MULTI_INDEX | NPY_ITER_REFS_OK,
+         NPY_KEEPORDER, NPY_NO_CASTING, NULL);
+
+     if (iter == NULL) {
+       return;
+     }
+
+     if (NpyIter_GetNDim(iter) != 2) {
+       // Assuming 2 dimensions for simplicity of example
+       return;
+     }
+
+     if (NpyIter_GetIterSize(iter) != 0) {
+       npy_intp *multi_index;
+       iternext = NpyIter_GetIterNext(iter, NULL);
+       if (iternext == NULL) {
+         NpyIter_Deallocate(iter);
+         return;
+       }
+
+       NpyIter_GetMultiIndexFunc *get_multi_index =
+           NpyIter_GetGetMultiIndex(iter, NULL);
+       if (get_multi_index == NULL) {
+         NpyIter_Deallocate(iter);
+         return;
+       }
+
+       dataptr = NpyIter_GetDataPtrArray(iter);
```

Comment:
I don't think it can fail, but just wanted to ask to make sure?

Reply:
Looks inconsistent in other examples / internally whether this is actually checked or not. I'll add for now - maybe something we clean up later in other places for consistency. Thanks